torch/ao/quantization/backend_config/qnnpack.py

```diff
@@ -11,17 +11,50 @@
     _get_rnn_op_configs,
     _get_share_qparams_op_configs,
 )
-from .backend_config import BackendConfig, DTypeConfig
+from .backend_config import BackendConfig, DTypeConfig, DTypeWithConstraints
 
 
 # ===================
 # |  DTYPE CONFIGS  |
 # ===================
 
-# TODO: For now, these DTypeConfigs are identical to the ones defined in native.py
-# In the future, once we support specifying quant_min/quant_max and scale_min/scale_max,
-# these will diverge. In particular, for QNNPACK, we will restrict the weight quantized
-# values to within [-127, 127] and set the min scale value to 2 ** -12.
+# xnnpack dtype configs
+
+# We restrict scale values to be 2 ** -12 to ensure the
+# requantization scale never falls below the xnnpack lower
+# threshold. Additionally, for qint8 weight, we restrict
+# the quantization values to [-127, +127], excluding -128.
+# For more detail, refer to the description of
+# `default_symmetric_qnnpack_qconfig`.
+
+# TODO: add additional restriction on qscheme to ensure it
+# is either per_tensor_symmetric or per_channel_symmetric
+
+xnnpack_act_qint8_with_constraints = DTypeWithConstraints(
+    dtype=torch.qint8,
+    scale_min_lower_bound=2 ** -12,
+)
+
+xnnpack_weight_qint8_with_constraints = DTypeWithConstraints(
+    dtype=torch.qint8,
+    quant_min_lower_bound=-127,
+    quant_max_upper_bound=127,
+    scale_min_lower_bound=2 ** -12,
+)
+
+xnnpack_weighted_op_qint8_dtype_config = DTypeConfig(
+    input_dtype=xnnpack_act_qint8_with_constraints,
+    output_dtype=xnnpack_act_qint8_with_constraints,
+    weight_dtype=xnnpack_weight_qint8_with_constraints,
+    bias_dtype=torch.float,
+)
+
+xnnpack_default_op_qint8_dtype_config = DTypeConfig(
+    input_dtype=xnnpack_act_qint8_with_constraints,
+    output_dtype=xnnpack_act_qint8_with_constraints,
+)
```

Comment:
cc @digantdesai do we have the min_scale constraints on normal ops as well? or just linear and conv?